resources/check-layout-th.js

```diff
@@ -249,4 +249,19 @@ window.checkLayout = function(selectorList, callDone = true)
         done();
 };
 
+// This function is a thin wrapper around `checkLayout` and simply reads the
+// `CHECK_LAYOUT_DELAY_MS` variable to add a delay. This global variable is not
+// intended to be set by browsers; instead, polyfills can set it to give
+// themselves time to apply changes before proceeding with assertions about the
+// layout. Tests that call this function at some point and do some other work
+// after should `await` it to avoid race conditions.
+window.checkLayoutDelayed = async function(selectorList, callDone = true) {
+    const delay = window.CHECK_LAYOUT_DELAY_MS;
+    if (typeof delay === 'number') {
+        console.log(`Waiting ${delay}ms before checking layout...`);
+        await new Promise(r => step_timeout(r, delay));
+    }
```

Comment:
@jerivas What do you think about the idea of ensuring this is not used when `--inject-polyfill` isn't used?

https://github.com/oddbird/wpt/pull/1#discussion_r1086022881

Reply:
+1 to the question about `--inject-polyfill`. Perhaps that should set another global that can be checked here?